xds/internal/client/bootstrap/bootstrap.go

```diff
@@ -182,6 +190,31 @@ func NewConfig() (*Config, error) {
 					serverSupportsV3 = true
 				}
 			}
+		case "certificate_providers":
+			var providerInstances map[string]json.RawMessage
+			if err := json.Unmarshal(v, &providerInstances); err != nil {
+				return nil, fmt.Errorf("xds: json.Unmarshal(%v) for field %q failed during bootstrap: %v", string(v), k, err)
+			}
+			configs := make(map[string]certprovider.StableConfig)
+			for instance, data := range providerInstances {
```

Comment:
What is an `instance`?

Reply:
A `plugin_instance`, which represents a `plugin_name`+`plugin_config`.

Reply:
This is probably minor detail.

Why does the `instance` need an "instance_name"? Would we have two instances with the same `plugin_name`, but different config? And since we always stop at the first valid config, what's the point?

This is similar to what we need for the balancing policy + config. We didn't make it a `map[string]{policy+config}`, but a `[]{policy+config}`.

Reply:
Yes, we can have two different instances with same `plugin_name` but different config. An `instance` is supposed to represent an instantiation of a plugin (which includes the `plugin_name`+`config`).

We don't stop at the first valid config here. We process all configs and use the `instance` specified by the control plane.